File: 2019/8/main.py
pixels = open('input.txt').read()

w, h = 25, 6
i = 0
layers = []
counts = []
fewest_zeros = 9999
answer_1 = None

while i < len(pixels) - 1:
    layer = {}
    counts = [0, 0, 0]
    for y in range(h):
        for x in range(w):
            pixel = int(pixels[i])
            layer[(x, y)] = pixel
            counts[pixel] += 1
            i += 1

    if counts[0] < fewest_zeros:
        fewest_zeros = counts[0]
        answer_1 = counts[1] * counts[2]

    layers.append(layer)

print(answer_1)

spirtes = [" ", "#", " "]

for y in range(h):
    for x in range(w):
        pixel = (l[x, y] for l in layers)
        p = list(filter(lambda x: x != 2, pixel))[0]
        print(spirtes[p], end="")
    print("")

File: 2019/8/main.hs
import Data.List.Split
import Data.List
import Data.Function


img :: Char -> Char
img c = if c == '1' then '#' else ' '
 
squash :: String -> Char
squash pixels = img $ head $ filter (/='2') pixels

paint :: [String] -> String
paint layers = unlines $ chunksOf 25 pixels
    where
        pixels  = map squash grouped
        grouped = [map (!!x) layers | x <- [0..(25 * 6)-1]]

main = do
    pixels <- readFile "input.txt"
    let layers = parmap concat $ chunksOf 6 $ chunksOf 25 pixels
    let answer1 = minimumBy (compare `on` (length.head)) $ map (group.sort) layers
    print $ product.tail $ map length answer1
    putStr $ paint layers
